### PR TITLE
feat: add large size for NumberInput

### PR DESCRIPTION
**Pull request by separatio**
[FX-2127]
[FX-2514]

### Description

This PR adds the large size variant for NumberInput.
Note: The `height` of the input should be 100% (24px), not 16px as it is presented in designs. This has been confirmed in the #-base-core channel.

### How to test

- check against designs in the ticket; here's [one](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/4f1f6493dfac89015cc6c71ea348807e931fe3bc/files/13531207-e094-44ec-ae1f-f27628c1aea5/layers/A85A9D7A-8D4B-4CC3-AEE8-224B1E4BB538?mode=build&selected=2522121955-AB221D70-1F22-40C3-A99B-899CB6482DF1&sha=d335874ff3dd8487bd8853d6e932ce2dbe23809b) of the designs that includes the large variant

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2127]: https://toptal-core.atlassian.net/browse/FX-2127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-2514]: https://toptal-core.atlassian.net/browse/FX-2514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by separatio** [comment]
I am aware that the visual screenshot looks bad. If you manually test it, however, it's perfectly fine.
I still have https://toptal-core.atlassian.net/browse/FX-2514 In Progress, that will completely fix the screenshot as well.

File: packages/picasso/src/NumberInput/NumberInput.tsx [hunk 1]
```diff
@@ -45,6 +46,7 @@ type NumberAdornmentProps = {
   inputRef: RefObject<HTMLInputElement>
   classes: Record<string, string>
   disabled?: boolean
+  size?: SizeType<'small' | 'medium' | 'large'>
```
Comment (denieler) [inline packages/picasso/src/NumberInput/NumberInput.tsx:49]: changeset says `added large size`, but here we have all 3 sizes actually appeared 😃

Comment (separatio) [reply packages/picasso/src/NumberInput/NumberInput.tsx:49]: That's for the NumberInputAdornment, not for NumberInput. And it fixes a bug, actually.

**Comment by separatio** [comment]
@toptal-bot run all

**Comment by separatio** [comment]
@toptal-bot run all

**Review by ozgurkececioglu — APPROVED** [review]
Tested and working just fine 👍

**Comment by separatio** [comment]
@toptal-bot run all

**Pull request merged**